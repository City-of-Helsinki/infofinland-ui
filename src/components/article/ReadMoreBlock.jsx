import Block from '@/components/layout/Block'
import { IconExternalSite } from '@/components/Icons'
import cls from 'classnames'

export const ExternalLinkCollection = ({ content = [], locale }) => (
  <div className="py-4">
    {content?.map(({ title, siteName, mainTranslation, languages }, i) => (
      <div
        className={cls({
          'mb-3 pb-3 border-b border-gray-hr': i + 1 < content.length,
        })}
        key={`readmore-${siteName}`}
      >
        <span
          className="flex items-center text-small"
          target="_blank"
          rel="noreferrer"
        >
          <IconExternalSite className="me-2" />
          {siteName}
        </span>

        <a
          rel="noreferrer"
          lang={mainTranslation?.field_locale}
          href={mainTranslation?.url}
          target="_blank"
          className="inline-block mb-2 text-body font-bold leading-snug ifu-text-link"
        >
          {title}
        </a>

        {languages?.length > 0 && (
          <div
            className={cls('flex flex-wrap divide-link divide-s', {
              lowercase: LOWERCASE_LOCALES.includes(locale),
            })}
          >
            {languages.map(({ url, title: langTitle, locale, id }, k) => (
              <a
                title={langTitle}
                rel="noreferrer"
                href={url}
                key={`link-${id}-${locale}`}
                lang={locale}
                className={cls('text-small leading-snug ifu-text-link pe-2', {
                  'ps-2': k > 0,
                })}
              >
                {langTitle}
              </a>
            ))}
          </div>
        )}
      </div>
    ))}
  </div>
)

const LOWERCASE_LOCALES = ['fi', 'et', 'es', 'fr']
const ReadMoreBlock = ({ content = [], locale }) => {
  if (content?.length === 0) {
    return null
  }
  return (
    <Block className="mt-8 mb-8 bg-orange-white">
      <ExternalLinkCollection content={content} locale={locale} />
    </Block>
  )
}

export default ReadMoreBlock
