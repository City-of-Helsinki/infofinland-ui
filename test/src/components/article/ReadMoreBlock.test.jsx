/**
 * @jest-environment jsdom
 */

import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import ReadMoreBlock from '@/components/article/ReadMoreBlock'
export const READMORE_CONTENT = [
  {
    siteName: 'DEMO:Kela',
    mainTranslation:{locale:'fi',url:'http://www.kela.fi/fi/'},
    pageName: 'Some random Kela page',
    languages: [
      { url: 'http://www.kela.fi/fi/', text: 'Suomi', lang: 'fi' },
      {
        url: 'http://www.kela.fi/sv/',
        text: 'Ruotsi',
        lang: 'sv',
      },

    ],
  },
]
describe('ReadMoreBlock', () => {
  it('should render the ReadMoreBlock', () => {
    const { asFragment } = render(<ReadMoreBlock content={READMORE_CONTENT}/>)
    expect(asFragment()).toMatchSnapshot()
  })
})
