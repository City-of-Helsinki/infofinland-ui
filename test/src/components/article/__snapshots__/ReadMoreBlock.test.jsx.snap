// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMoreBlock should render the ReadMoreBlock 1`] = `
<DocumentFragment>
  <div
    class="px-6 lg:px-12 lg:mx-12  xl:mx-28 2xl:mx-48  3xl:ms-64  3xl:max-w-4xl mt-8 mb-8 bg-orange-white"
  >
    <div
      class="py-6"
    >
      <div
        class=""
      >
        <span
          class="flex items-center text-small"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="inline-block me-2"
            fill="none"
            height="10"
            viewBox="0 0 10 10"
            width="10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.88889 8.88889H1.11111V1.11111H5V0H1.11111C0.494444 0 0 0.5 0 1.11111V8.88889C0 9.5 0.494444 10 1.11111 10H8.88889C9.5 10 10 9.5 10 8.88889V5H8.88889V8.88889ZM6.11111 0V1.11111H8.10556L2.64444 6.57222L3.42778 7.35556L8.88889 1.89444V3.88889H10V0H6.11111Z"
              fill="black"
            />
          </svg>
          DEMO:Kela
        </span>
        <a
          class="inline-block mb-4 text-body font-bold leading-snug ifu-text-link"
          href="http://www.kela.fi/fi/"
          lang="fi"
          rel="noreferrer"
          target="_blank"
        >
          Some random Kela page
        </a>
        <div
          class="flex flex-wrap divide-link divide-s lowercase"
        >
          <a
            class="text-small leading-snug ifu-text-link pe-2"
            href="http://www.kela.fi/fi/"
            rel="noreferrer"
            title="Some random Kela page"
          >
            Suomi
          </a>
          <a
            class="text-small leading-snug ifu-text-link pe-2 ps-2"
            href="http://www.kela.fi/sv/"
            rel="noreferrer"
            title="Some random Kela page"
          >
            Ruotsi
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
